src/QsCompiler/CompilerOptimizations/PureCircuitAPI.fs

```diff
@@ -4,197 +4,175 @@
 module Microsoft.Quantum.QsCompiler.CompilerOptimization.PureCircuitAPI
 
 open System.Collections.Immutable
-open System.Numerics
 open Microsoft.Quantum.QsCompiler.DataTypes
 open Microsoft.Quantum.QsCompiler.SyntaxTokens
 open Microsoft.Quantum.QsCompiler.SyntaxTree
 
 open ComputationExpressions
-open Types
 open Utils
 open Printer
 
 
-/// An identifier for a qubit or for an array of qubits
-type QubitReference = Qubit of int | QubitArray of int
-
 /// Any constant expression
-type Const =
-| UnitValue
-| ValueTuple            of Const list
+type Literal =
 | IntLiteral            of int64
-| BigIntLiteral         of BigInteger
 | DoubleLiteral         of double
-| BoolLiteral           of bool
-| StringLiteral         of string
-| ResultLiteral         of QsResult
 | PauliLiteral          of QsPauli
-| RangeLiteral          of Const * Const
-| ValueArray            of Const list
-| ArrayItem             of Const * Const
-| QubitReference        of QubitReference
+| PauliArray            of QsPauli list
+
+/// Any expression
+type Expression =
+| Literal               of Literal
+| Tuple                 of Expression list
+| Qubit                 of int
+| QubitArray            of int list
+| UnknownValue          of int
 
 /// A call to a quantum gate, with the given functors and arguments
 type GateCall = {
     gate:     QsQualifiedName
     adjoint:  bool
-    controls: Const list
-    arg:      Const
+    controls: Expression list
```

Comment:
What Expressions exactly are you allowing here? Why is this an Expression list opposed to just an Expression?
A doc comment would be helpful.

Reply:
This should be an int list instead of an Expression list, as the only valid control is a qubit array. Fixed in latest commit.